### PR TITLE
feature(CB2-5701): Create the TRL Brakes Component

### PR DESCRIPTION
**Pull request by tahayazici**
## Amend TRL Brakes section for a provisional record

[CB2-5701](https://dvsa.atlassian.net/browse/CB2-5701)

### CODE REVIEW

File: src/app/forms/components/base-control/base-control.component.ts [hunk 1]
```diff
@@ -31,6 +33,9 @@ export class BaseControlComponent implements ControlValueAccessor, AfterContentI
   ngAfterContentInit(): void {
     const ngControl: NgControl | null = this.injector.get(NgControl, null);
     if (ngControl) {
+      if (!ngControl.control) {
```
Comment (Ian-Hodges) [inline src/app/forms/components/base-control/base-control.component.ts:36]: Was this just for local debugging, or is it useful in production?

Comment (tahayazici) [reply src/app/forms/components/base-control/base-control.component.ts]: Local. Removed. Thanks 🙂

File: src/app/forms/components/brakes/brakes.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ngOnInit(): void {
+    this.template = TrlBrakes;
+
+    this.form = this.dfs.createForm(this.template, this.vehicleTechRecord) as CustomFormGroup;
+
+    this.form.cleanValueChanges
+      .pipe(debounceTime(400), takeUntil(this.destroy$))
+      .subscribe(e => this.formChange.emit(e));
+  }
+
```
Comment (naathanbrown) [inline src/app/forms/components/brakes/brakes.component.ts:37]: Does this not need the patch values on change thing too?

Comment (tahayazici) [reply src/app/forms/components/brakes/brakes.component.ts:37]: Yes, it does need it. Thanks. Added.

**Review by Ian-Hodges — APPROVED** [review]

**Pull request merged**